rct229/rulesets/ashrae9012019/section19/section19rule21.py

```diff
@@ -0,0 +1,498 @@
+                ):
+                    UNDETERMINED_MSG = f"The baseline system {hvac_id_b} supply air cfm >= 5,000 CFM and the OA fraction is >= 70 %OA, per G3.1.2.10 exhaust air energy recovery is required to be modeled in the baseline and it has not been modeled. However, the system serves kitchen type spaces and exception 3 may be applicable which is that commercial kitchen hoods (grease) classified as Type 1 by NFPA 96 shall not require exhaust air energy recovery to be modeled if exhaust air energy recovery is not used in the proposed design (note that it has not modeled in the proposed). Conduct manual check to determine if exception 3 is applicable."
+                else:
+                    UNDETERMINED_MSG = f"Exhaust air energy recovery was not modeled in the baseline when the OA fraction is greater or equal to 70% and the supply cfm is greater or equal to 5,000 cfm. However, it appears that Section 90.1 G3.1.2.10 Exception 6 may be applicable which states, 'Where the largest exhaust source is less than 75% of the design outdoor airflow. This exception shall only be used if exhaust air energy recovery is not used in the proposed design.'. There are multilple exhaust fans associated with the hvac system and therefore this could not be assessed as the configuration is unknown. Conduct manual check to determine if this exception applies. If not, then fail."
+
+                return UNDETERMINED_MSG
+
+            def rule_check(self, context, calc_vals=None, data=None):
+                climate_zone = calc_vals["climate_zone"]
+                OA_fraction_b = calc_vals["OA_fraction_b"]
+                supply_airflow_b = calc_vals["supply_airflow_b"]
+                ER_modeled_b = calc_vals["ER_modeled_b"]
+                exception_1_applies = calc_vals["exception_1_applies"]
+                exception_2_applies = calc_vals["exception_2_applies"]
+                exception_6_applies = calc_vals["exception_6_applies"]
+                exception_7_applies = calc_vals["exception_7_applies"]
+                ER_not_req_for_heating_sys = calc_vals["ER_not_req_for_heating_sys"]
+                sys_type_heating_only = calc_vals["sys_type_heating_only"]
+                all_lighting_space_types_defined = calc_vals[
+                    "all_lighting_space_types_defined"
+                ]
+                all_ventilation_space_types_defined = calc_vals[
+                    "all_ventilation_space_types_defined"
+                ]
+                ER_modeled_p = calc_vals["ER_modeled_p"]
+
+                return (
+                    (
+                        OA_fraction_b >= OA_fraction_b_70
+                        and supply_airflow_b >= SUPPLY_AIRFLOW_5000CFM
+                        and (
+                            # CASE 1
+                            (ER_modeled_b)
+                            or
+                            # CASE 2
+                            (not ER_modeled_b and exception_1_applies)
```

Comment:
We actually don't need the `not ER_modeled_b` here.
The reason is because if `ER_modeled_b` is true, the logic is going to short-circuit and return True.
Therefore, if the logic ever moved to CASE 2 or other CASES, it is guaranteed that `ER_modeled_b == False`

Reply:
Good point! Addressed.